bin/src/commands.rs

```diff
@@ -20,3 +26,46 @@ pub async fn ui_error(message: String, _stack: Option<Vec<String>>) -> AppResult
 
     Ok(())
 }
+
+#[tauri::command]
+pub async fn add_contract(
+    chain_id: u64,
+    address: Address,
+    db: tauri::State<'_, Db>,
+) -> AppResult<()> {
+    let networks = ethui_networks::Networks::read().await;
+
+    let provider = async move {
+        match networks.get_network(chain_id as u32) {
+            Some(network) => Ok(network.get_alloy_provider().await?),
+            _ => Err(Error::InvalidNetwork(chain_id as u32)),
+        }
+    }
```

Comment:
why isn't this just `networks.get_network(...)?.get_alloy_provider().await?`

Reply:
Not quite, `get_network` returns an Option, not a Result, so we can't use the `?`

Reply:
so you can use `get_network().ok_or(Error::InvalidNetwork(chain_id as u32))`